lib/ex_aws/ecs.ex

```diff
@@ -1021,16 +1049,48 @@ defmodule ExAws.ECS do
 
   ## Private parts
 
-  # Pascalize has room for improvements.. but it seems to work
+  defp transform_list(opts, list_key, value_key, fn_transform) do
+    transformed = opts
+    |> Keyword.get(list_key, [])
+    |> transform_list(value_key, fn_transform)
 
-  defp pascalize_keys(opts) when is_map(opts) do
+    Keyword.put(opts, list_key, transformed)
+  end
+  defp transform_list(list, value_key, fn_transform) do
+    list
+    |> Enum.map(&(transform_value(&1, value_key, fn_transform)))
+  end
+
+  defp transform_value(map, key, fn_transform) when is_map(map) do
+    map
+    |> Keyword.new
+    |> transform_value(key, fn_transform)
+    |> Map.new
+  end
+  defp transform_value(opts, key, fn_transform) do
     opts
-    |> Enum.map(&pascalize_keys/1)
-    |> Enum.into(%{})
+    |> Keyword.get_and_update(key, fn(value) ->
+        new_val = fn_transform.(value)
+        {value, new_val}
+      end)
+    |> update_only_if_existed(opts)
+  end
+
+  defp update_only_if_existed({nil, _}, original), do: original
+  defp update_only_if_existed({_, transformed}, _), do: transformed
+
+  defp replace_underscore_with_dash(value) when is_atom(value) do
+    value
+    |> Atom.to_string
+    |> replace_underscore_with_dash
+  end
+  defp replace_underscore_with_dash(value) when is_binary(value), do: String.replace(value, "_", "-")
+
+  defp pascalize_keys(opts) when is_map(opts) do
+    Map.new(opts, &pascalize_keys/1)
   end
-  defp pascalize_keys([]), do: []
-  defp pascalize_keys([head | tail]) do
-    [pascalize_keys(head)] ++ pascalize_keys(tail)
+  defp pascalize_keys(list) when is_list(list) do
+    list |> Enum.map(&pascalize_keys/1)
   end
   defp pascalize_keys(bool) when is_boolean(bool), do: bool
   defp pascalize_keys(atom) when is_atom(atom), do: Atom.to_string(atom)
```

Comment:
Booleans are in fact atoms, and we want `true` and  `false` as values, not `"true"` and  `"false"`.  However, I have not handle the case when the value is `:true` or `:false`, just realizing that they are booleans, and values will be passed on. 
I've never seen booleans as those atoms, but suspect that the json-converters will handle them correctly so we don't need to match on them.